internal/uvm/share.go

```diff
@@ -0,0 +1,75 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/requesttype"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// Share shares in file(s) from `reqHostPath` on the host machine to `reqUVMPath` inside the UVM.
+// This function handles both LCOW and WCOW scenarios.
+func (uvm *UtilityVM) Share(ctx context.Context, reqHostPath, reqUVMPath string, readOnly bool) (retErr error) {
+	if uvm.OS() == "windows" {
+		if _, err := uvm.GetVSMBUvmPath(ctx, reqHostPath, readOnly); err == ErrNotAttached {
+			// share file has not been added yet, add it now
+			options := uvm.DefaultVSMBOptions(readOnly)
+			vsmbShare, err := uvm.AddVSMB(ctx, reqHostPath, options)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				if retErr != nil {
```

Comment:
Can we use `err` for the return value or do we need the different name for some reason?

Reply:
I just used `retErr` so I didn't get confused but it's not needed